pkg/model/core/route.go

```diff
@@ -0,0 +1,331 @@
+package core
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	gateway_api_v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gateway_api_v1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+type Route interface {
+	GetSpec() RouteSpec
+	GetStatus() RouteStatus
+	GetName() string
+	GetNamespace() string
+	GetDeletionTimestamp() *v1.Time
+	GetRuntimeObject() runtime.Object
```

Comment:
Fair point, will get rid of this noise!

Reply:
My concern here is that the actual members have the same name as the function then. If I have a variable like so:
```go
var httpRoute *core.HTTPRoute = // Some HTTPRoute
httpRoute.Name // Is this the Name variable or the Name function?
```

I would rather keep the Get to avoid such conflicts

Reply:
Because the property is a consequence of embedding and the getter is necessary for the interface. If we just expose the embedded struct, we break the abstraction and lose the common behaviours.

Reply:
Spoke with mikhail offline, his concern was about making the property private instead, which is a great idea. Will fix next commit!